scientific/src/commonMain/kotlin/formatter/CommonFormatter.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ Copyright 2022 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.scientific.formatter
+
+import com.splendo.kaluga.scientific.ScientificValue
+import com.splendo.kaluga.scientific.unit.ScientificUnit
+import kotlin.math.roundToLong
+
+private typealias CustomFormatHandler = (Number) -> String
+sealed class CommonFormatter(builder: Builder) : Formatter {
+    class Builder {
+        companion object {
+            fun build(build: Builder.() -> Unit = {}): Formatter {
+                val builder = Builder()
+                build(builder)
+                return Buildable(builder)
+            }
+        }
+
+        internal val customFormatters = mutableMapOf< ScientificUnit<*>, CustomFormatHandler>()
+        fun ifUnitIs(unit: ScientificUnit<*>, format: CustomFormatHandler) {
+            customFormatters[unit] = format
+        }
```

Comment:
Why is it better? Not necessary against it, but I guess original naming provides nice human-readable DSL.
Look hob this builder usage looks like: 
```
formatter = CommonFormatter.Builder.build {
            ifUnitIs(Kilometer per Hour) { "$it Kilometer per Hour" }
            ifUnitIs(Liter) { "$it Liter" }
}
```

So `if the unit is __Kilometer per Hour__ use "$ Kilometer per Hour" format`


After your change it will be:
```
formatter = CommonFormatter.Builder.build {
            updateFormatForUnit(Kilometer per Hour) { "$it Kilometer per Hour" }
            updateFormatForUnit(Liter) { "$it Liter" }
}
```

I even not talking about that you are not updating formatter but saying builder to use formater for a unit. I mean `update`  word in the name  is definitely wrong as it is not representing the action it does

Reply:
Alternatively can use something like this b41080e17a03926e78e998ddaad34c885d91a4ae

Reply:
here with infix b59e4caa0f12d72fab4fa3c6958315ab8a3edf27